provider/pkg/gen/schema.go

```diff
@@ -451,6 +455,97 @@ func (g *packageGenerator) genResource(typeName string, createMethod, getMethod,
 	return nil
 }
 
+// buildIdParams creates a map of parameters that are needed to build resource IDs from an ID path template.
+// Keys are API parameter names, values are SDK property names (that may be equal to keys, or not).
+func (g *packageGenerator) buildIdParams(typeName string, idPath string, inputProperties map[string]schema.PropertySpec, response *discovery.JsonSchema) (map[string]string, error) {
+	result := map[string]string{}
+
+	subMatches := pathRegex.FindAllStringSubmatch(idPath, -1)
+	for idx, names := range subMatches {
+		name := names[1]
+
+		// If the property is already defined in the input args, add its SDK name.
+		sdkName := apiNameToSdkName(name)
+		if _, has := inputProperties[sdkName]; has {
+			result[name] = sdkName
+			continue
+		}
+
+		// If the property is already defined in the response, add its API name.
+		// Note: API name equals to SDK name for all response properties, currently.
+		if _, has := response.Properties[name]; has {
+			result[name] = name
+			continue
+		}
+
+		// The last parameter represents the resource name. It is not a part of the create path,
+		// and isn't always modelled as an explicit property in the SDKs. If it's missing in the SDK,
+		// it's assigned by the API, and the provider needs to infer its value from the response.
+		isLast := idx == len(subMatches)-1
+		if !isLast {
+			return nil, errors.Errorf("property %q not found in %q", name, idPath)
+		}
+
+		// First, check if we have an explicit annotation for which property to use for name resolution.
+		key := fmt.Sprintf("%s:%s.%s", g.mod, typeName, name)
+		if v, has := resourceNamePropertyOverrides[key]; has {
+			if !g.schemaContainsProperty(response, v) {
+				return nil, errors.Errorf("property %q not found in response schema of %q", v, key)
+			}
+			result[name] = v
+			return result, nil
```

Comment:
Curious - why is this better? I actually changed from `continue`. Not saying `return` is better but trying to learn.

Reply:
It was just a tiny bit alarming to see an early (non error) return from within the iteration through the matches during the initial scan. Might also be something that got drilled into my head from "Code Complete" years ago. I didn't notice the assertion on the last element initially. 

This might be just me though :).

Reply:
I see, good point. I added a comment to explain the return - hopefully it's the best of two options.